### PR TITLE
feat: add MiniAppDirectJump

### PR DESCRIPTION
**Pull request by Xunflash**
# feat: add MiniAppDirectJump

## Description

长按小程序/分享卡片增加直接打开APP的菜单项，省去打开卡顿的小程序/网页的步骤（暂时仅支持哔哩哔哩）
## Issues Fixed or Closed by This PR
（（回滚失误不小心把pr给close了，顺便直接把嵌套的if改了，另外其他两个问题是从qa里面抄的代码，应该问题不大吧（？
## Check List

- [x] I have tested the changes and verified that they work and don't break anything(as well as I can manage) or drop the support for previous versions.
- [x] I have merged commits that are meaningless for follow-up work and confirmed that they will not cause damage to follow-up maintenance

### CODE REVIEW

File: app/src/main/java/top/xunflash/hook/MiniAppDirectJump.kt [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+            if (!m.returnType.isArray) {
+                continue
+            }
+            val ps = m.parameterTypes
+            if (ps.size == 1 && ps[0] == View::class.java) {
+                XposedBridge.hookMethod(m, getMenuItemCallBack)
+                break
+            }
+        }
+    }
```
Comment (qwq233) [inline app/src/main/java/top/xunflash/hook/MiniAppDirectJump.kt:88]:
```suggestion
private fun expandShortUrl(shortUrl: String): String? = CoroutineScope(Dispatchers.IO).future {
        val connection = (URL(shortUrl).openConnection() as HttpURLConnection).apply {
            instanceFollowRedirects = false
            connectTimeout = 10000
            readTimeout = 10000
        }
        val result = connection.getHeaderField("Location")
        connection.disconnect()
        result
    }.get()
```

**Review by qwq233 — CHANGES_REQUESTED** [review]

File: app/src/main/java/top/xunflash/hook/MiniAppDirectJump.kt (hunk 1 above)
Comment (qwq233) [reply app/src/main/java/top/xunflash/hook/MiniAppDirectJump.kt:88]: 一般来讲这可能有点阻塞，你看需求改

Comment (qwq233) [reply app/src/main/java/top/xunflash/hook/MiniAppDirectJump.kt:88]: 哦 这个是说你哪个什么expandShortUrl的 括的地方错了

Comment (Xunflash) [reply app/src/main/java/top/xunflash/hook/MiniAppDirectJump.kt:91]: 已修改

**Comment by qwq233** [comment]
你改了测试过没有的，主要我写的这些随便写的，没测试的

**Comment by Xunflash** [comment]
> 你改了测试过没有的，主要我写的这些随便写的，没测试的

测试了，没问题的

**Review by qwq233 — APPROVED** [review]

**Pull request merged**